### DISCUSSION TITLE
gatsby-ffmpeg throws errors and kicks me out of dev session when querying with graphql

### DISCUSSION THREAD

**Discussion by Eric-Alain** [gatsbyjs/gatsby#31535] [Help] [answered]
## Description

I want to serve a locally hosted video in an optimized format, so naturally, I looked to the gatsby plugins. I followed the instructions on the following page: [https://www.gatsbyjs.com/plugins/gatsby-video/](https://www.gatsbyjs.com/plugins/gatsby-video/)

I can't figure out why it's doing this. Ultimately, I am just looking for some way to serve a locally hosted video, in an optimized format and preferably through graphql. I am not attached to this method though, so any thoughts are welcome.

### Steps to reproduce

Repo: A gatsby starter template with a video component where I make attempt to make a graphql query for a video. 

Steps: 

- npm install --save gatsby-video gatsby-transformer-ffmpeg gatsby-plugin-ffmpeg
- add `gatsby-transformer-ffmpeg` to plugins in gatsby-config.js
- make video component that makes a static query

### Expected result

The query should return the video src either in the dev environment or in the graphql tool (http://localhost:8000/___graphql)

### Actual result

Throws the following error and then terminates my gatsby develop session: 

C:\Users\erica\Documents\Web Writing\bug-repro\node_modules\yoga-layout-prebuilt\yoga-layout\build\Release\nbind.js:53
        throw ex;
        ^

TypeError: Cannot read property 'createJob' of undefined
    at C:\Users\erica\Documents\Web Writing\bug-repro\node_modules\gatsby-plugin-ffmpeg\lib\index.js:175:27
    at C:\Users\erica\Documents\Web Writing\bug-repro\node_modules\gatsby-plugin-ffmpeg\lib\index.js:141:3
    at C:\Users\erica\Documents\Web Writing\bug-repro\node_modules\gatsby-plugin-ffmpeg\node_modules\async\queue.js:10:5
    at Object.process (C:\Users\erica\Documents\Web Writing\bug-repro\node_modules\gatsby-plugin-ffmpeg\node_modules\async\internal\queue.js:239:17)
    at C:\Users\erica\Documents\Web Writing\bug-repro\node_modules\gatsby-plugin-ffmpeg\node_modules\async\internal\queue.js:97:19
    at Immediate.<anonymous> (C:\Users\erica\Documents\Web Writing\bug-repro\node_modules\gatsby-plugin-ffmpeg\node_modules\async\internal\setImmediate.js:17:41)
    at processImmediate (internal/timers.js:462:21)

### Environment

Gatsby info:

  System:
    OS: Windows 10 10.0.19042
    CPU: (8) ia32 AMD Ryzen 7 3700U with Radeon Vega Mobile Gfx  
  Binaries:
    Node: 14.17.0 - C:\Program Files (x86)\nodejs\node.EXE
    npm: 7.13.0 - C:\Program Files (x86)\nodejs\npm.CMD
  Languages:
    Python: 3.9.5 - /c/Python39/python
  Browsers:
    Chrome: 90.0.4430.212
    Edge: Spartan (44.19041.964.0), Chromium (90.0.818.62)
  npmPackages:
    gatsby: ^3.4.1 => 3.4.1 
    gatsby-plugin-ffmpeg: ^0.3.1 => 0.3.1 
    gatsby-plugin-gatsby-cloud: ^2.4.1 => 2.4.1 
    gatsby-plugin-image: ^1.4.0 => 1.4.0 
    gatsby-plugin-manifest: ^3.4.0 => 3.4.0 
    gatsby-plugin-offline: ^4.4.0 => 4.4.0 
    gatsby-plugin-react-helmet: ^4.4.0 => 4.4.0 
    gatsby-plugin-sharp: ^3.4.1 => 3.4.1 
    gatsby-source-filesystem: ^3.4.0 => 3.4.0 
    gatsby-transformer-ffmpeg: ^0.3.8 => 0.3.8 
    gatsby-transformer-sharp: ^3.4.0 => 3.4.0 
    gatsby-video: ^0.2.0 => 0.2.0 
  npmGlobalPackages:
    gatsby-cli: 3.5.0

Flags: 

PARALLEL_SOURCING: true

**Comment by vladar** [comment, marked as answer]
My best guess is that this plugin is a bit outdated and may be incompatible with gatsby v3? I recommend you to open an issue in the correct repository though as we do not maintain it: https://github.com/Mike-Dax/gatsby-plugin-ffmpeg

**Reply by Eric-Alain** [reply]
Thanks, I appreciate it, I am new to creating github issues. I will do that.
